pkg/kn/commands/types_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright © 2019 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type typeTestCase struct {
+	kubeCfgPath  string
+	precedence   []string
+	explicitPath string
+}
+
+func TestGetClientConfig(t *testing.T) {
```

Comment:
Nice work @edwardstudy. Consider also using the new `gotest.tools` which should make your tests cleaner, especially assertions and dividing into multiple parts. We agreed during yesterday's WG meeting to move all tests using this small addition to `gotest.tools`. See #134's tests for examples or let's chat today in Beijing, I am working on some. Cheers :beers: